### PR TITLE
Replace dword with u32

### PR DESCRIPTION
**Pull request by Plotja**
#23

### CODE REVIEW

File: src/xrCore/cpuid.cpp [hunk 1]
```diff
@@ -174,7 +174,7 @@ unsigned int query_processor_info(processor_info* pinfo)
 
 #ifdef XR_PLATFORM_WINDOWS
     DWORD returnedLength = 0;
```
Comment (Xottab-DUTY) [inline src/xrCore/cpuid.cpp:176]:
```suggestion
    u32 returnedLength = 0;
```

File: src/xrNetServer/empty/NET_Client.cpp [hunk 2]
```diff
@@ -175,7 +175,7 @@ XRNETSERVER_API bool psNET_direct_connect = false;
  ****************************************************************************/
 
 static HRESULT WINAPI Handler(PVOID pvUserContext,
-    DWORD dwMessageType,
+    u32 dwMessageType,
```
Comment (Xottab-DUTY) [inline src/xrNetServer/empty/NET_Client.cpp]: WinAPI function, should use WinAPI types.
```suggestion
    DWORD dwMessageType,
```

File: src/xrPhysics/xrPhysics.cpp [hunk 3]
```diff
@@ -24,7 +24,7 @@ static void load(int argc, char** argv, char** envp)
 }
 
 #if defined(XR_PLATFORM_WINDOWS)
-BOOL APIENTRY DllMain(HMODULE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
+BOOL APIENTRY DllMain(HMODULE hModule, u32 ul_reason_for_call, LPVOID lpReserved)
```
Comment (Xottab-DUTY) [inline src/xrPhysics/xrPhysics.cpp]: WinAPI function, should use WinAPI types.
```suggestion
BOOL APIENTRY DllMain(HMODULE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
```

File: src/xrCore/xrCore.cpp [hunk 4]
```diff
@@ -431,9 +431,9 @@ void xrCore::CalculateBuildId()
 
 #if defined(XR_PLATFORM_WINDOWS)
 #ifdef _EDITOR
-BOOL WINAPI DllEntryPoint(HINSTANCE hinstDLL, DWORD ul_reason_for_call, LPVOID lpvReserved)
+BOOL WINAPI DllEntryPoint(HINSTANCE hinstDLL, u32 ul_reason_for_call, LPVOID lpvReserved)
 #else
-BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD ul_reason_for_call, LPVOID lpvReserved)
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, u32 ul_reason_for_call, LPVOID lpvReserved)
```
Comment (Xottab-DUTY) [inline src/xrCore/xrCore.cpp]: WinAPI function, should use WinAPI types.
```suggestion
BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD ul_reason_for_call, LPVOID lpvReserved)
```

File: src/xrCore/xrCore.cpp [hunk 5]
```diff
@@ -431,9 +431,9 @@ void xrCore::CalculateBuildId()
 
 #if defined(XR_PLATFORM_WINDOWS)
 #ifdef _EDITOR
-BOOL WINAPI DllEntryPoint(HINSTANCE hinstDLL, DWORD ul_reason_for_call, LPVOID lpvReserved)
+BOOL WINAPI DllEntryPoint(HINSTANCE hinstDLL, u32 ul_reason_for_call, LPVOID lpvReserved)
```
Comment (Xottab-DUTY) [inline src/xrCore/xrCore.cpp]: WinAPI function, should use WinAPI types.
```suggestion
BOOL WINAPI DllEntryPoint(HINSTANCE hinstDLL, DWORD ul_reason_for_call, LPVOID lpvReserved)
```

File: src/xrGame/RegistryFuncs.cpp [hunk 6]
```diff
@@ -117,13 +117,13 @@ void WriteRegistry_StrValue(LPCSTR rKeyName, const char* value)
     WriteRegistryValue(rKeyName, REG_SZ, value);
 #endif
 }
-void ReadRegistry_DWValue(LPCSTR rKeyName, DWORD& value)
+void ReadRegistry_DWValue(LPCSTR rKeyName, u32& value)
```
Comment (Xottab-DUTY) [inline src/xrGame/RegistryFuncs.cpp]: Uses WinAPI, should use WinAPI types.
```suggestion
void ReadRegistry_DWValue(LPCSTR rKeyName, DWORD& value)
```

File: src/xrGame/RegistryFuncs.cpp [hunk 7]
```diff
@@ -117,13 +117,13 @@ void WriteRegistry_StrValue(LPCSTR rKeyName, const char* value)
...
 {
 #ifndef XR_PLATFORM_LINUX // FIXME!!!
     ReadRegistryValue(rKeyName, REG_DWORD, &value);
 #endif
 }
-void WriteRegistry_DWValue(LPCSTR rKeyName, const DWORD& value) 
+void WriteRegistry_DWValue(LPCSTR rKeyName, const u32& value)
```
Comment (Xottab-DUTY) [inline src/xrGame/RegistryFuncs.cpp]: Uses WinAPI, should use WinAPI types.
```suggestion
void WriteRegistry_DWValue(LPCSTR rKeyName, const DWORD& value) 
```

File: src/xrGame/RegistryFuncs.h [hunk 8]
```diff
@@ -6,5 +6,5 @@ void WriteRegistry_StrValue(LPCSTR rKeyName, const char* value);
 u32 const ReadRegistry_BinaryValue(LPCSTR rKeyName, u8* buffer_dest, u32 const buffer_size);
 void WriteRegistry_BinaryValue(LPCSTR rKeyName, u8 const* buffer_src, u32 const buffer_size);
 
-void ReadRegistry_DWValue(LPCSTR rKeyName, DWORD& value);
-void WriteRegistry_DWValue(LPCSTR rKeyName, const DWORD& value);
+void ReadRegistry_DWValue(LPCSTR rKeyName, u32& value);
+void WriteRegistry_DWValue(LPCSTR rKeyName, const u32& value);
```
Comment (Xottab-DUTY) [inline src/xrGame/RegistryFuncs.h]: Uses WinAPI, should use WinAPI types.
```suggestion
void ReadRegistry_DWValue(LPCSTR rKeyName, DWORD& value);
void WriteRegistry_DWValue(LPCSTR rKeyName, const DWORD& value);
```

File: src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp [hunk 9]
```diff
@@ -259,12 +259,12 @@ SVS* CResourceManager::_CreateVS(LPCSTR _name)
         {
             if (pShaderBuf)
             {
-                _hr = HW.pDevice->CreateVertexShader((DWORD*)pShaderBuf->GetBufferPointer(), &_vs->vs);
+                _hr = HW.pDevice->CreateVertexShader((u32*)pShaderBuf->GetBufferPointer(), &_vs->vs);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp]: Should use DWORD.
```suggestion
                _hr = HW.pDevice->CreateVertexShader((DWORD*)pShaderBuf->GetBufferPointer(), &_vs->vs);
```

File: src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp [hunk 10]
```diff
@@ -259,12 +259,12 @@ SVS* CResourceManager::_CreateVS(LPCSTR _name)
...
                 if (SUCCEEDED(_hr))
                 {
                     LPCVOID data = NULL;
                     _hr = D3DXFindShaderComment(
-                        (DWORD*)pShaderBuf->GetBufferPointer(), MAKEFOURCC('C', 'T', 'A', 'B'), &data, NULL);
+                        (u32*)pShaderBuf->GetBufferPointer(), MAKEFOURCC('C', 'T', 'A', 'B'), &data, NULL);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp]: Should use DWORD.
```suggestion
                        (DWORD*)pShaderBuf->GetBufferPointer(), MAKEFOURCC('C', 'T', 'A', 'B'), &data, NULL);
```

File: src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp [hunk 11]
```diff
@@ -387,12 +387,12 @@ SPS* CResourceManager::_CreatePS(LPCSTR name)
         {
             if (pShaderBuf)
             {
-                _hr = HW.pDevice->CreatePixelShader((DWORD*)pShaderBuf->GetBufferPointer(), &_ps->ps);
+                _hr = HW.pDevice->CreatePixelShader((u32*)pShaderBuf->GetBufferPointer(), &_ps->ps);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp]: Should use DWORD.
```suggestion
                _hr = HW.pDevice->CreatePixelShader((DWORD*)pShaderBuf->GetBufferPointer(), &_ps->ps);
```

File: src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp [hunk 12]
```diff
@@ -387,12 +387,12 @@ SPS* CResourceManager::_CreatePS(LPCSTR name)
...
                 if (SUCCEEDED(_hr))
                 {
                     LPCVOID data = NULL;
                     _hr = D3DXFindShaderComment(
-                        (DWORD*)pShaderBuf->GetBufferPointer(), MAKEFOURCC('C', 'T', 'A', 'B'), &data, NULL);
+                        (u32*)pShaderBuf->GetBufferPointer(), MAKEFOURCC('C', 'T', 'A', 'B'), &data, NULL);
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderDX9/dx9ResourceManager_Resources.cpp]: Should use DWORD.
```suggestion
                        (DWORD*)pShaderBuf->GetBufferPointer(), MAKEFOURCC('C', 'T', 'A', 'B'), &data, NULL);
```

File: src/utils/ETools/ETools.cpp [hunk 13]
```diff
@@ -5,7 +5,7 @@
 
 #pragma warning(disable : 4267)
 
-BOOL APIENTRY DllMain(HANDLE hModule, DWORD fdwReason, LPVOID lpReserved)
+BOOL APIENTRY DllMain(HANDLE hModule, u32 fdwReason, LPVOID lpReserved)
```
Comment (Xottab-DUTY) [inline src/utils/ETools/ETools.cpp]: WinAPI function, should use DWORD.
```suggestion
BOOL APIENTRY DllMain(HANDLE hModule, DWORD fdwReason, LPVOID lpReserved)
```

File: src/utils/xrDXT/Image_DXTC.cpp [hunk 14]
```diff
@@ -26,11 +26,11 @@ from the use or inability to use this file or items derived from it.
 // should be in ddraw.h
 #ifndef MAKEFOURCC
 #define MAKEFOURCC(ch0, ch1, ch2, ch3)\
-    ((DWORD)(BYTE)(ch0) | ((DWORD)(BYTE)(ch1) << 8) | ((DWORD)(BYTE)(ch2) << 16) | ((DWORD)(BYTE)(ch3) << 24))
+    ((u32)(BYTE)(ch0) | ((u32)(BYTE)(ch1) << 8) | ((u32)(BYTE)(ch2) << 16) | ((u32)(BYTE)(ch3) << 24))
 #endif // defined(MAKEFOURCC)
 
 // Defined in this module:
-WORD GetNumberOfBits(DWORD dwMask);
+WORD GetNumberOfBits(u32 dwMask);
```
Comment (Xottab-DUTY) [inline src/utils/xrDXT/Image_DXTC.cpp]:
```suggestion
u16 GetNumberOfBits(u32 dwMask);
```

File: src/utils/xrDXT/Image_DXTC.cpp [hunk 15]
```diff
@@ -976,7 +976,7 @@ DXTColBlock* pBlock, Color8888* col_0, Color8888* col_1, Color8888* col_2, Color
 // Desc: Returns the number of bits set in a DWORD mask
 //	from microsoft mssdk d3dim sample "Compress"
 //-----------------------------------------------------------------------------
-WORD GetNumberOfBits(DWORD dwMask)
+WORD GetNumberOfBits(u32 dwMask)
```
Comment (Xottab-DUTY) [inline src/utils/xrDXT/Image_DXTC.cpp]:
```suggestion
u16 GetNumberOfBits(u32 dwMask)
```

File: src/utils/xrDXT/Image_DXTC.cpp [hunk 16]
```diff
@@ -976,7 +976,7 @@ DXTColBlock* pBlock, Color8888* col_0, Color8888* col_1, Color8888* col_2, Color
...
 {
     WORD wBits = 0;
```
Comment (Xottab-DUTY) [inline src/utils/xrDXT/Image_DXTC.cpp]:
```suggestion
    u16 wBits = 0;
```

File: src/utils/xrLC_Light/xrLC_Light.cpp [hunk 17]
```diff
@@ -13,7 +13,7 @@ b_params& g_params()
     return inlc_global_data()->g_params();
 }
 
-BOOL APIENTRY DllMain(HMODULE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
+BOOL APIENTRY DllMain(HMODULE hModule, u32 ul_reason_for_call, LPVOID lpReserved)
```
Comment (Xottab-DUTY) [inline src/utils/xrLC_Light/xrLC_Light.cpp]: WinAPI function, should use DWORD.
```suggestion
BOOL APIENTRY DllMain(HMODULE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
```

File: src/utils/xrSE_Factory/xrSE_Factory.cpp [hunk 18]
```diff
@@ -32,7 +32,7 @@ FACTORY_API void __stdcall destroy_entity(IServerEntity*& abstract)
 }
 };
 
-BOOL APIENTRY DllMain(HANDLE module_handle, DWORD call_reason, LPVOID reserved)
+BOOL APIENTRY DllMain(HANDLE module_handle, u32 call_reason, LPVOID reserved)
```
Comment (Xottab-DUTY) [inline src/utils/xrSE_Factory/xrSE_Factory.cpp]: WinAPI function, should use DWORD.
```suggestion
BOOL APIENTRY DllMain(HANDLE module_handle, DWORD call_reason, LPVOID reserved)
```

File: src/xrCore/cpuid.cpp (hunk 1 above)
Comment (Plotja) [reply src/xrCore/cpuid.cpp:176]: returnedLength is a argument to WinApi GetLogicalProcessorInformation function. It should be DWORD

Comment (Xottab-DUTY) [reply src/xrCore/cpuid.cpp:176]: Aa... Well, you already accepted the suggested change, so, it's fine.

**Pull request merged**